connector-pagerduty/src/test/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyTestUtils.java

```diff
@@ -159,29 +176,97 @@ private static JsonObject getClientLink(final JsonObject oldInnerPayload, String
 
         return clientLink;
     }
+}
 
-    private static JsonObject getSourceNames(final JsonObject oldInnerSourceNames) {
-        if (oldInnerSourceNames != null) {
-            JsonObject newInnerSourceNames = new JsonObject();
+/**
+ * This class emulates the functionality of the same class within notifications-engine.
+ */
+class InsightsUrlsBuilder {
```

Comment:
why dupplicating this class instead of using `engine/src/main/java/com/redhat/cloud/notifications/processors/InsightsUrlsBuilder.java` ?
if it is just to force the env url, you can use mokito like `Mockito.when(environment.url()).thenReturn(DEFAULT_ENVIRONMENT_URL);` on relevant tests.

Reply:
The environment URL isn't the issue, [this duplicate actually hardcodes it](https://github.com/RedHatInsights/notifications-backend/blob/7c24129395315c53b457287f17bda22f88f00bca/connector-pagerduty/src/test/java/com/redhat/cloud/notifications/connector/pagerduty/PagerDutyTestUtils.java#L240).

The PagerDuty connector currently does not have a test dependency on `notifications-engine`, so I can't import InsightsUrlsBuilder and have to duplicate the class instead. If it's not a big deal, I can just add the dependency and use that suggestion.

Reply:
Ok sorry I didn't noticced it was in two distinct modules.
Then if engine is in charge to compute application and inventory url, why should we compute them again in pagerDuty connector, those url should be some simple input payload text properties like other fieds? WDYT?

Reply:
Sure, I'll hardcode the fields. This was more useful when I was validating the logic for the connector.